src/jinja2/runtime.py

```diff
@@ -152,6 +152,8 @@ def resolve_or_missing(context, key, missing=missing):
         return context.vars[key]
     if key in context.parent:
         return context.parent[key]
+    if key in context.environment.filters:
```

Comment:
Wasn't too sure about this since this will allow users to use filters as variables, which was previously not a thing. However, i believe this is similar to how python deals with built-ins and overriding them.